src/hosted-buttons/types.js

```diff
@@ -1,8 +1,27 @@
 /* @flow */
 /* eslint-disable no-restricted-globals, promise/no-native */
 
+export type FlexDirection = string;
+export type Color = string;
+export type Layout = string;
+
+export type FundingSources = string;
+export interface GetFlexDirection {
+  flexDirection: FlexDirection;
+}
+
+export interface GetFlexDirectionArgs {
+  layout: Layout;
+}
+
+export interface BuildButtonContainerArgs {
+  flexDirection: FlexDirection;
+  selector: string | HTMLElement;
+}
+
 export type HostedButtonsComponentProps = {|
   hostedButtonId: string,
+  fundingSources?: $ReadOnlyArray<FundingSources>,
```

Comment:
Since we default this to `[]` should it be not optional?
```suggestion
  fundingSources: $ReadOnlyArray<FundingSources>,
```

Reply:
I originally had it as an optional prop but added the default empty array to avoid some type errors. I will update.